src/test/java/seedu/guest/model/guest/RoomTest.java

```diff
@@ -25,20 +25,24 @@ public void isValidRoom() {
         assertThrows(NullPointerException.class, () -> Room.isValidRoom(null));
 
         // invalid room
-        assertFalse(Room.isValidRoom("")); // empty string
-        assertFalse(Room.isValidRoom(" ")); // spaces only
+        assertFalse(Room.isValidRoom("")); // empty string, near boundary values
+        assertFalse(Room.isValidRoom(" ")); // spaces only, near boundary values
         assertFalse(Room.isValidRoom("-50-23")); // not allowed to start with a hyphen
         assertFalse(Room.isValidRoom("50 - 23")); // not allowed to have spacings in between
         assertFalse(Room.isValidRoom("50--23")); // not allowed to have more than 1 hyphen in between
         assertFalse(Room.isValidRoom("$50")); // '$' is not allowed
         assertFalse(Room.isValidRoom("^&*")); // special symbols only are not allowed
         assertFalse(Room.isValidRoom("aloha-5()")); // '(' or ')' is not allowed
+        assertFalse(Room.isValidRoom("A".repeat(101))); // above 100 characters is not allowed (near boundary values)
 
         // valid room
         assertTrue(Room.isValidRoom("05-73")); // only alphanumeric characters and hyphen
         assertTrue(Room.isValidRoom("RoomA")); // only alphanumeric characters
+        assertTrue(Room.isValidRoom("1")); // 1 character is allowed (near boundary values)
         assertTrue(Room.isValidRoom("Room5")); // hyphens are optional
         assertTrue(Room.isValidRoom("Aloha-5")); // only alphanumeric characters and hyphen
+        assertTrue(Room.isValidRoom("A".repeat(100))); // 100 characters is allowed (near boundary values)
+        assertTrue(Room.isValidRoom("A".repeat(80))); // less than 100 characters is allowed
```

Comment:
Don't think we need this as we have test cases with less than 100 characters already!

Reply:
Sounds good!